src/vmm/src/device_manager/mmio.rs

```diff
@@ -249,11 +263,18 @@ impl MMIODeviceManager {
         mmio_device: MmioTransport,
         _cmdline: &mut kernel_cmdline::Cmdline,
     ) -> Result<MMIODeviceInfo> {
-        let mmio_slot = self.allocate_new_slot(1)?;
-        self.register_mmio_virtio(vm, device_id, mmio_device, &mmio_slot)?;
+        let mmio_device_info;
+        {
+            mmio_device_info = MMIODeviceInfo {
+                addr: self.allocate_slot(MMIO_LEN)?,
+                len: MMIO_LEN,
+                irqs: self.allocate_irq(vm, &mmio_device.locked_device().interrupt_evt())?,
+            };
+        }
```

Comment:
Can't this be shortened?
```suggestion
        let mmio_device_info = MMIODeviceInfo {
            addr: self.allocate_slot(MMIO_LEN)?,
            len: MMIO_LEN,
            irqs: self.allocate_irq(vm, &mmio_device.locked_device().interrupt_evt())?,
        };
```